### ISSUE TITLE
Invisible scheme for danger buttons is missing

### ISSUE THREAD

**Issue by HDinger** [primer/view_components#2969]
# Pain point

* We often use invisble schemes of buttons to avoid a cognitive overload in an area where multiple buttons are shown next to each other.
* The `danger` scheme of buttons is a great thing to hint users of a potentially dangerous action. This effect is achieved by a red font and the strong hover effect. However, there is no invisible scheme for the danger buttons which breaks the design heavily.
* As the danger effect is only achieved via font and on hover, it could be possible to create an `invisible_danger` scheme (although I have to admit that the name might be confusing).

## Example screenshot 
**With icon buttons**
<img width="133" alt="Bildschirmfoto 2024-07-26 um 08 15 02" src="https://github.com/user-attachments/assets/ca49bd1a-5303-4be4-815e-307578e3e362">

I know that this is basically a deisgn decision, so I'd be interested in your opinions on this. I'd also be happy to create a PR for such a scheme if you agree with that approach.

**Comment by camertron** [comment]
Thanks for bringing this up! Imma let Primer Design weigh in on this 😄

**Comment by tallys** [comment]
We'll bring this up at an upcoming Primer Patterns, there might be some issues around the accessibility — we generally do not use color to indicate meaning on icons. 

cc @mperrotti can you add this to the agenda?

**Comment by mperrotti** [comment]
We discussed this in today's Primer Patterns working session and we don't actually think this button needs to be a `"danger"` variant. For destructive or irremediable actions, we typically show a confirmation dialog for extra friction.

TL;DR - just use the `"invisible"` variant and show a confirmation dialog if it's actually a destructive action.

**Comment by lesliecdubs** [comment]
👋 I'm going to go ahead and close this issue as per the recommendation in the last comment: https://github.com/primer/view_components/issues/2969#issuecomment-2271924797. Please feel free to respond for clarification, or reopen the issue if there's more you'd like to discuss. Thanks!
